### DISCUSSION TITLE
Disable docusaurus router for specific route

### DISCUSSION THREAD

**Discussion by danielleroux** [facebook/docusaurus#8563] [Q&A]
Hi,

is it possible to exclude a specific route<strike>r</strike> from the docusaurus router?
I need to host a small web app (react, angular) inside the static folder, but the router blocks the react-router from the react app.

```
<repo>/docs/test.md    <-- <iframe src="/my-app/custom/app/route">...
<repo>/static/my-app   <-- dist folder from my-app react build (configured route 'custom/app/route')
```

**Comment by slorber** [comment]
I don't know what you mean by "exclude specific router from Docusaurus router" or "but the router blocks the react-router from the react app".

If you need to navigate to an external React app hosted on the same domain, you can try linking to it through the `pathname://` protocol:
https://docusaurus.io/docs/advanced/routing#escaping-from-spa-redirects

If you want to embed a React app inside an iframe, clicking on links inside the iframe will not change the main URL of your browser. This is how the web works and not a Docusaurus-specific behavior.

**Reply by danielleroux** [reply]
Thanks for your answer please checkout the following:
https://stackblitz.com/edit/github-xaddn4?file=src%2Fpages%2Findex.js

Home Page (src/pages/index.js):

```tsx
export default function Home() {
  const { siteConfig } = useDocusaurusContext();
  return (
    <Layout
      title={`Hello from ${siteConfig.title}`}
      description="Description will go into a meta tag in <head />"
    >
      <main>
        <iframe
          src="/webapp"
          style={{ width: '100%', height: '20rem' }}
        ></iframe>
        <iframe
          src="/webapp/test/route"
          style={{ width: '100%', height: '20rem' }}
        ></iframe>
      </main>
    </Layout>
  );
}
```

Inside the static folder is a react app with the following routes configured:

```tsx
import React from 'react';
import ReactDOM from 'react-dom/client';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import App from './App';
import './index.css';

const router = createBrowserRouter([
  {
    path: '/webapp/',
    element: <App />,
  },
  {
    path: '/webapp/test/route',
    element: <div>Test Route 1</div>,
  },
]);

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>
);
```
If you go to `<container url>/webapp/` you see the react app but if you go to `<container url>/webapp/test/route` you get the docusaurus error page.

**Comment by danielleroux** [comment]
@slorber i dont want to change links inside my app this is working for sure and has nothing to do with docusaurus. But as i wrote in the answer need to specifiy a iframe with a link to the react app inside static folder:

```html
        <iframe
          src="/webapp" <--- Works react app is shown
          style={{ width: '100%', height: '20rem' }}
        ></iframe>    
        <iframe
          src="/webapp/test/route" <--- Dont work 404 from docusaurus is shown
          style={{ width: '100%', height: '20rem' }}
        ></iframe>
```

**Reply by slorber** [reply]
Your test route doesn't work even outside of Docusaurus so it's unlikely to work when embedded as an iframe.

<img width="802" alt="CleanShot 2023-01-25 at 16 34 18@2x" src="https://user-images.githubusercontent.com/749374/214605953-9fb8f1e0-8339-419b-83e0-77c2720f071c.png">


Most likely this is because there's no static file at `static/webapp/test/route.html` or `static/webapp/test/route/index.html`, so Docusaurus serves its `build/404.html` page and never loads your app.

You have to make your web server (dev+prod) serve `static/webapp/index.html` or `static/webapp/404.html` for all requests under `/webapp/*`. 

Hosts (and modern local dev servers too) usually do this by default for the root static deployment, but not for subpaths, and you have to configure this yourself if you have multiple apps served inside the same deployment. 

For example with Netlify you can try to put in `_redirects`: `/webapp/* /webapp/index.html 200`

That's outside the scope of Docusaurus: you have to make your custom SPA work with your local/prod host first before thinking about embedding it in another site.
